test/e2e/gatewayapi_upgradeable_test.go

```diff
@@ -0,0 +1,124 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	gatewayapi_upgradeable "github.com/openshift/cluster-ingress-operator/pkg/operator/controller/gatewayapi-upgradeable"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
+	"github.com/openshift/cluster-ingress-operator/pkg/manifests"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func kubeinit() {
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		panic(err)
+	}
+	kclient, err = client.New(kubeConfig, client.Options{})
+	if err != nil {
+		panic(err)
+	}
+}
+
+func TestGatewayAPIUpgradeable(t *testing.T) {
+	t.Parallel()
+
+	//t.Log("Starting test: Ensure the Gateway API CRDs exist")
+	createCRDs(t)
+
+	//t.Log("Starting test: Testing if admin gate works properly")
+	testAdminGate(t, true)
+
+	//t.Log("Starting test: Deleting Gateway API CRDs")
+	deleteCRDs(t)
+
+	//t.Log("Starting test: Testing if admin gate works properly after CRDs are deleted")
+	testAdminGate(t, false)
+}
+
+func createCRDs(t *testing.T) {
+	crds := []*apiextensionsv1.CustomResourceDefinition{
+		manifests.GatewayClassCRD(),
+		manifests.GatewayCRD(),
+		manifests.HTTPRouteCRD(),
+		manifests.ReferenceGrantCRD(),
+	}
+
+	for _, crd := range crds {
+		if err := kclient.Create(context.TODO(), crd); err != nil && !errors.IsAlreadyExists(err) {
+			t.Fatalf("Failed to create CRD %s: %v", crd.Name, err)
+		} else if errors.IsAlreadyExists(err) {
+			//t.Logf("CRD %s already exists", crd.Name)
+		} else {
+			//t.Logf("Successfully created CRD %s", crd.Name)
+		}
+	}
+}
+
+func testAdminGate(t *testing.T, shouldExist bool) {
+	t.Helper()
+	crdsExist := checkCRDsExist(t)
+	configMap := &corev1.ConfigMap{}
+	err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 30*time.Second, false, func(ctx context.Context) (bool, error) {
+		if err := kclient.Get(ctx, client.ObjectKey{Namespace: "openshift-config-managed", Name: "admin-gates"}, configMap); err != nil {
+			//t.Logf("Failed to get configmap admin-gates: %v", err)
+			if errors.IsNotFound(err) {
+				//t.Logf("ConfigMap 'admin-gates' not found in namespace openshift-config-managed")
+			} else {
+				//t.Logf("Error getting ConfigMap: %v", err)
+			}
```

Comment:
This log message, on the other hand, does make sense as it signals a problem:
```suggestion
			t.Logf("Failed to get configmap admin-gates: %v, retrying...", err)
```
And yes, you don't need to make a distinction between error types because the error will contain the needed description. E.g. if it's "NotFound" error, the description will say so.